### PR TITLE
REF: Add explicit visibility check in move refactoring

### PR DESCRIPTION
**Pull request by dima74**
[`RsImportHelper` can find private path](https://github.com/intellij-rust/intellij-rust/issues/4000), so we have to explicitly check accessibility of such paths

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/refactoring/move/common/RsMoveUtil.kt [hunk 1]
```diff
@@ -217,6 +221,7 @@ fun <T : RsElement> movedElementsDeepDescendantsOfType(
  */
 fun RsVisRestriction.updateScopeIfNecessary(psiFactory: RsPsiFactory, newParent: RsMod) {
     if (crateRoot == newParent.crateRoot) {
+        if (path.textNormalized == "self") return  // `pub(self)`
```
Comment (ortem) [inline src/main/kotlin/org/rust/ide/refactoring/move/common/RsMoveUtil.kt:224]: Maybe use `PathKind.SELF` instead of string literal?

**Review by ortem — APPROVED** [review]
LGTM except one comment

**Comment by dima74** [comment]
bors r=ortem

**Comment by dima74** [comment]
bors ping

**Comment by dima74** [comment]
bors retry

**Pull request merged**